matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/RoomAPI.kt

```diff
@@ -233,11 +234,22 @@ internal interface RoomAPI {
     ): SendResponse
 
     /**
-     * Get state events of a room
+     * Get all state events of a room
      * Ref: https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-rooms-roomid-state
      */
     @GET(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/state")
-    suspend fun getRoomState(@Path("roomId") roomId: String): List<Event>
+    suspend fun getAllRoomStates(@Path("roomId") roomId: String): List<Event>
+
+    /**
+     * Get specific state event of a room
+     * Ref: https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey
+     */
+    @GET(NetworkConstants.URI_API_PREFIX_PATH_R0 + "rooms/{roomId}/state/{eventType}/{state_key}")
+    suspend fun getRoomState(
+                @Path("roomId") roomId: String,
+                @Path("eventType") eventType: String,
+                @Path("state_key") stateKey: String
+    ): PinnedEventsStateResponse
```

Comment:
Expecting a return type of `PinnedEventsStateResponse` is not generic enough for this API. The returned type should be `Content`